### ISSUE TITLE
Different type behavior when importing

### ISSUE THREAD

**Issue by MatthewChang** [facebook/flow#3664]
I've come across a confusing behavior testing out flowtype(0.43.0) on a redux project, using the react-redux typing from the flow-typed repo.

```
// @flow
import React from 'react'
import {connect} from 'react-redux'
class ChildComponent extends React.Component {
  props: {
    value: string, 
    otherValue: string
  }
  render() {
    return <span>{ this.props.value && this.props.otherValue }</span>
  }
}
function mapStateToProps(state) {
  return {otherValue: "test"}
}
const ConnectedChildComponent = connect(mapStateToProps)(ChildComponent)
class ParentComponent extends React.Component {
  render() {
    return <ConnectedChildComponent/>
  }
}
```

The above code performs the type inference as expected and reports an error in the usage in the parent, indicating that the "value" prop should be set.

However, when I move the container child code into a separate file I find that the typecheck does not happen (likely to to an implicit 'any' import)

```
// flowTestPatent.js
// @flow
import React from 'react'
import {connect} from 'react-redux'
import ConnectedChildComponent from './flowTestChild'
class ParentComponent extends React.Component {
  render() {
    return <ConnectedChildComponent/>
  }
}


//flowTestChild.js
// @flow
import React from 'react'
import {connect} from 'react-redux'

export class ChildComponent extends React.Component {
  props: {
    value: string, 
    otherValue: string
  }
  render() {
    return <span>{ this.props.value && this.props.otherValue }</span>
  }
}
function mapStateToProps(state) {
  return {otherValue: "test"}
}
export default connect(mapStateToProps)(ChildComponent)
```

The above code reports no errors when it seems like it should be equivalent to the first example. I would assume I am somehow performing the import incorrectly, but when I import the un-connected component instead, typechecking happens as expected. Can anybody help explain what's going on here? Thanks!

**Comment by chrisvasz** [comment]
Same problem here, with the exact same symptoms. The unconnected component typechecks correctly and the connected component does not.

**Comment by cbranch101** [comment]
I'm running into the same issue when importing `compose` from `recompose` It works in the same directory, but not when exported

**Comment by gnprice** [comment]
This issue can be closed, because it was fixed in v0.143.0.

The cause of the symptoms above is that the react-redux libdef [at that time](https://github.com/flow-typed/flow-typed/blob/bdf1e66252cd340dc64526edbab97f1e63d10ef6/definitions/npm/react-redux_v5.x.x/flow_v0.30.x-/react-redux_v5.x.x.js#L1) began with the following line:
```js
import type { Dispatch, Store } from 'redux'
```

That's an import at the top level of a libdef file, outside any `declare module` block. This has always been invalid… and until v0.143.0 Flow's behavior was to give no error where it happens, and instead to give either very confusing errors (like above) elsewhere in your codebase, or else silently no errors at all where related types are used.

Since v0.143.0 Flow gives a nice clear error when a libdef does this, and when you move the import inside the `declare module` block all is well. See #6414 for details.
